setup-env

```diff
@@ -17,50 +17,89 @@ It does the following:
   - Configures git upstream remote "lineage" repositories.
 
 Usage:
-  setup-env [options] [virt_env_name]
-  setup-env (-h | --help)
+  setup-env [-n] [virt_env_name]
+  setup-env [-v] [python_version]
+  setup-env (-h)
 
 Options:
-  -f --force          Delete virtual enviroment if it already exists.
-  -h --help           Show this message.
-  -i --install-hooks  Install hook environments for all environments in the
-                      pre-commit config file.
+  -f      Delete virtual enviroment if it already exists.
+  -h      Show this message.
+  -i      Install hook environments for all environments in the
+          pre-commit config file.
+  -l      List available Python versions and select interactively.
+  -n      Choose the name of the virtual environment.
+  -v      Specify the Python version for the virtual environment.
```

Comment:
Since we make a point of using long options wherever possible, I don't think we should drop them here just because `getopt` on BSD (used by MacOS) doesn't support them.  Instead of using `getopt`, how about implementing something along the lines of [this](https://github.com/SixArm/getoptsy/blob/main/getoptsy) or [this](https://stackoverflow.com/a/28466267)?

Reply:
@dav3r I will look into these and get back with you

Reply:
@dav3r - I think a better option is to use GNU `getopt`, which does support long options.

Reply:
I'm fine with that, but I think you already knew that. :smiley:

Reply:
Well I was making some headway with the suggestions that @dav3r made. I was just working through a few buggy issues. But if we did want to make it a dependency I could maybe put in steps to install it in the setup-env script

Reply:
Just for context I have `--force` `--list-versions` `--help` and `--install-hooks` working but the `--version [arg]` and `--name [arg]` are giving me some trouble but still figuring those out

Reply:
LGTM!

Reply:
Viva los long options!